tests/by-util/test_du.rs

```diff
@@ -1580,6 +1580,41 @@ fn test_du_inodes_total_text() {
     assert!(parts[0].parse::<u64>().is_ok());
 }
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[test]
+fn test_du_summary_total_mega_duplicates() {
+    // TODO: Add checks for non-Linux platforms
+    let ts = TestScenario::new_fresh(util_name!());
+    let at = &ts.fixtures;
+
+    ts.cmd("dd")
+        .args(&["if=/dev/urandom", "of=file1", "bs=1M", "count=3"])
+        .succeeds();
+    at.mkdir("dir1");
+    ts.cmd("dd")
+        .args(&["if=/dev/urandom", "of=dir1/file1", "bs=1M", "count=5"])
+        .succeeds();
+    at.mkdir("dir2");
+    ts.cmd("dd")
+        .args(&["if=/dev/urandom", "of=dir2/file1", "bs=1M", "count=7"])
+        .succeeds();
+
+    let result = ts.ucmd().args(&["-smc", "dir1", "."]).succeeds();
+
+    let result_reference = unwrap_or_return!(expected_result(&ts, &["-smc", "dir1", "."]));
+    if result_reference.succeeded() {
+        assert_eq!(result.stdout_str(), result_reference.stdout_str());
+        return;
+    }
+
+    du_summary_total_mega_duplicates(result.stdout_str());
+}
+
+fn du_summary_total_mega_duplicates(s: &str) {
+    // TODO: Add checks for non-Linux platforms
+    assert_eq!(s, "5\tdir1\n10\t.\n15\ttotal\n");
+}
```

Comment:
This function causes a "dead code" warning on mac and Windows. To avoid the warning, I would either use the same `cfg` as you use  for `test_du_summary_total_mega_duplicates` or move the `assert_eq!` to the test function and remove `du_summary_total_mega_duplicates`.